common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/ImageIndexableTest.scala

```diff
@@ -0,0 +1,155 @@
+package uk.ac.wellcome.pipeline_storage
+
+import com.sksamuel.elastic4s.Index
+import org.scalatest.Assertion
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import uk.ac.wellcome.elasticsearch.test.fixtures.ElasticsearchFixtures
+import uk.ac.wellcome.fixtures.TestWith
+import uk.ac.wellcome.models.Implicits._
+import uk.ac.wellcome.models.work.generators.{ImageGenerators, WorkGenerators}
+import uk.ac.wellcome.models.work.internal
+import uk.ac.wellcome.models.work.internal.{AugmentedImage, SourceWorks}
+import uk.ac.wellcome.models.work.internal.SourceWork._
+import uk.ac.wellcome.pipeline_storage.Indexable.imageIndexable
+import uk.ac.wellcome.pipeline_storage.fixtures.ElasticIndexerFixtures
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class ImageIndexableTest
+    extends AnyFunSpec
+    with ScalaFutures
+    with Matchers
+    with ElasticsearchFixtures
+    with ElasticIndexerFixtures
+    with ImageGenerators
+    with WorkGenerators {
+
+  describe("updating images with merged / redirected sources") {
+    it("overrides an image with the same version if source version is higher") {
+      val source = createIdentifiedWork
+      val originalImage = createAugmentedImageWith(
+        parentWork = source,
+        redirectedWork = None,
+        version = 2
+      )
+      val updatedSourceImage = originalImage.copy(
+        source = internal.SourceWorks(
+          canonicalWork = source.toSourceWork.copy(version = 2),
+          redirectedWork = None
+        )
+      )
+
+      withImagesIndexAndIndexer {
+        case (index, indexer) =>
+          val insertFuture = ingestInOrder(indexer)(
+            originalImage,
+            updatedSourceImage
+          )
+
+          whenReady(insertFuture) { result =>
+            assertIngestedImageIs(
+              result = result,
+              ingestedImage = updatedSourceImage,
+              index = index
+            )
+          }
+      }
+    }
```

Comment:
It took me a moment to follow this test, because there are two versions going around. (FWIW, the storage-service has explicit types for versions to make this easier.)

Minor notes:

* Does it matter what the version of the original image is? `version = 2` looks significant, but I’m not sure why.
* How do you know the version of `source` is 1? It is because the default on `createIdentifiedWork` happens to be 1, but that's not immediately obvious reading the test.

   We care that it increases in the updated image; the exact value is less important. I would prefer something like:

    ```scala
    val updatedWork = sourceWork.toSourceWork.copy(version = sourceWork.version + 1)
    ```